packages/myst-cli/src/transforms/citations.ts

```diff
@@ -25,55 +24,42 @@ function pushCite(
   };
 }
 
-function addCitationChildren(
-  cite: Cite,
-  renderer: CitationRenderer,
-  kind: CiteKind = 'parenthetical',
-): boolean {
+function addCitationChildren(cite: Cite, renderer: CitationRenderer): boolean {
   const render = renderer[cite.label as string];
+  if (!render) {
+    cite.error = 'not found';
+    return false;
+  }
+  let children: StaticPhrasingContent[];
   try {
-    const children = render?.inline(kind === 'narrative' ? InlineCite.t : InlineCite.p, {
+    children = render.inline(cite.kind === 'narrative' ? InlineCite.t : InlineCite.p, {
       prefix: cite.prefix,
       suffix: cite.suffix,
+      partial: cite.partial,
     }) as StaticPhrasingContent[];
-    if (children) {
-      cite.children = children;
-      return true;
-    }
   } catch (error) {
-    // pass
+    cite.error = 'rendering error';
+    return false;
   }
-  cite.error = true;
-  return false;
+  if (!hasChildren(cite)) cite.children = children;
+  delete cite.error;
+  return true;
 }
 
 function hasChildren(node: Parent) {
   return node.children && node.children.length > 0;
 }
 
 export function transformCitations(
-  log: Logger,
   mdast: GenericParent,
   renderer: CitationRenderer,
   references: Pick<References, 'cite'>,
-  file: string,
 ) {
-  // TODO: this can be simplified if typescript doesn't die on the parent
-  const citeGroups = selectAll('citeGroup', mdast) as CiteGroup[];
-  citeGroups.forEach((node) => {
```

Comment:
I guess we had this before to pass `citeGroup` node `kind` to the `cite` children...? I don't see any equivalent now; am I missing something or do we just not need it? It shouldn't matter for the new `@cite` stuff, but does it matter for the old unchanged `cite` role code where `kind` is only on the `citeGroup`...?

Reply:
I think that we previously only kept the "parenthetical" information on the cite-group, and now that is in both places, meaning that we don't have to first do the groups, then individual nodes (as well as skipping some).

This is both equivalent and a bit faster probably. Simpler from the code perspective as well.